streaming/src/test/scala/org/apache/spark/streaming/StreamingContextSuite.scala

```diff
@@ -297,6 +299,23 @@ class StreamingContextSuite extends SparkFunSuite with BeforeAndAfter with Timeo
     Thread.sleep(100)
   }
 
+  test("registering and de-registering of streamingSource") {
+    val conf = new SparkConf().setMaster(master).setAppName(appName)
+    ssc = new StreamingContext(conf, batchDuration)
+    addInputStream(ssc).register()
+
+    ssc.start()
+    assert(ssc.getState() === StreamingContextState.INITIALIZED)
+    assert(StreamingContextSuite.sources.get(StreamingContextSuite.streamingSource)!= "null")
```

Comment:
Makes sense. I'll write that up. Was having problems when I initially wrote as it was in ExecutionManagerSuite. I'll figure out something.

Thank you.